contracts/utils/StaticDelegateCall.sol

```diff
@@ -0,0 +1,38 @@
+// SPDX-License-Identifier: MIT
+
+pragma solidity ^0.8.0;
+
+/**
+ * @dev Provides emulation of static delegatecalls for implementing view-only
+ * functions that execute another contract's code with the calling contract's
+ * context.
+ */
+abstract contract StaticDelegateCall {
```

Comment:
Can this be a library? Every file in /utils is and I think this fits into that same category.

Reply:
It can't because it relied on the contract exposing `delegateCallAndRevert` publicly.

Reply:
This one was modeled after `Multicall`, but I can look into it. I had a few questions below.

Reply:
Just realized Multicall is an abstract contract as well, so that'd make a case for keeping this one abstract.
I'll answer your questions in each thread.